webdriver/tests/bidi/browsing_context/set_viewport/invalid.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+
+import webdriver.bidi.error as error
+
+
+@pytest.mark.parametrize("value", [None, False, 42, {}, []])
+@pytest.mark.asyncio
+async def test_params_context_invalid_type(bidi_session, value):
+    with pytest.raises(error.InvalidArgumentException):
+        await bidi_session.browsing_context.set_viewport(context=value, viewport={
+            "width": 100,
+            "height": 200,
+        })
+
+
+@pytest.mark.asyncio
+async def test_params_context_invalid_value(bidi_session):
+    with pytest.raises(error.NoSuchFrameException):
+        await bidi_session.browsing_context.set_viewport(context="_invalid_")
+
+
+@pytest.mark.parametrize("viewport", [{"width": None, "height": 300}, {"width": 1200, "height": None}, {"width": 1200, "height": "String"}])
+@pytest.mark.asyncio
```

Comment:
There are different checks that we have to perform.

1. types other than `{}` for the viewport itself.
2. an empty object, or only height or width
3. a loop for all the invalid values for height and width checked separately while giving a valid value for the other field.

With all that we won't need a parameterization of the test.

Reply:
See my comment https://github.com/web-platform-tests/wpt/pull/40584#discussion_r1238613000 (Python does not allow sending those cases given how I implemented the client)

Reply:
Added those cases.